python/ray/data/datasource/binary_datasource.py

```diff
@@ -24,27 +23,21 @@ class BinaryDatasource(FileBasedDatasource):
 
     _COLUMN_NAME = "bytes"
 
-    def _read_file(self, f: "pyarrow.NativeFile", path: str, **reader_args):
-        from pyarrow.fs import HadoopFileSystem
+    def __init__(
+        self,
+        paths: Union[str, List[str]],
+        include_paths: bool = False,
+        **file_based_datasource_kwargs,
+    ):
+        super().__init__(paths, **file_based_datasource_kwargs)
 
-        include_paths = reader_args.pop("include_paths", False)
-        if reader_args.get("compression") == "snappy":
-            import snappy
+        self.include_paths = include_paths
 
-            filesystem = reader_args.get("filesystem", None)
-            rawbytes = BytesIO()
-
-            if isinstance(filesystem, HadoopFileSystem):
-                snappy.hadoop_snappy.stream_decompress(src=f, dst=rawbytes)
-            else:
-                snappy.stream_decompress(src=f, dst=rawbytes)
-
-            data = rawbytes.getvalue()
```

Comment:
The current `_read_file` logic is:

> We'll try to decompress the file for you, but we can't decompress Snappy files. So, if you try to read a Snappy file, we'll pass in a `compression` argument in `reader_args`.

I've simplified the interface by moving the Snappy handling logic to `FileBasedDatasource`:

>  We'll give you the decompressed file regardless of compression.